File: Nx3CS/bin/test/src/nx3/PPartComplexesGenerator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PPartComplexesGenerator : global::haxe.lang.HxObject {
		public    PPartComplexesGenerator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PPartComplexesGenerator(global::nx3.PPart part){
			unchecked {
				#line 18 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				global::nx3.PPartComplexesGenerator.__hx_ctor_nx3_PPartComplexesGenerator(this, part);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PPartComplexesGenerator(global::nx3.PPartComplexesGenerator __temp_me112, global::nx3.PPart part){
			unchecked {
				#line 20 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				__temp_me112.part = part;
				__temp_me112.vvoices = part.getVoices();
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				return new global::nx3.PPartComplexesGenerator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				return new global::nx3.PPartComplexesGenerator(((global::nx3.PPart) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PPart part;
		
		public  global::Array<object> vvoices;
		
		public  global::Array<object> complexes;
		
		public virtual   global::Array<object> getComplexes(){
			unchecked {
				#line 28 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				if (( this.complexes != default(global::Array<object>) )) {
					#line 28 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
					return this.complexes;
				}
				
				#line 29 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				this.positionsMap = this.calcPositionsMap();
				this.calcComplexes(this.positionsMap);
				return this.complexes;
			}
			#line default
		}
		
		
		public virtual   void calcComplexes(global::haxe.ds.IntMap<object> positions){
			unchecked {
				#line 53 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				this.complexes = new global::Array<object>(new object[]{});
				global::Array<int> poskeys = global::cx.MapTools.keysToArray<int>(positions.keys());
				poskeys = global::cx.MapTools.sortarray<int>(poskeys);
				{
					#line 56 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
					int _g = 0;
					#line 56 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
					while (( _g < poskeys.length )){
						#line 56 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						int pos = poskeys[_g];
						#line 56 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						 ++ _g;
						#line 58 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						global::Array<object> vnotes = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (positions.@get(pos).@value) ))) );
						global::nx3.PComplex vcomplex = new global::nx3.PComplex(((global::nx3.PPart) (this.part) ), ((global::Array<object>) (vnotes) ), ((int) (pos) ));
						this.complexes.push(vcomplex);
					}
					
				}
				
			}
			#line default
		}
		
		
		public  global::haxe.ds.IntMap<object> positionsMap;
		
		public virtual   global::haxe.ds.IntMap<object> calcPositionsMap(){
			unchecked {
				#line 69 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				global::haxe.ds.IntMap<object> positionsMap = new global::haxe.ds.IntMap<object>();
				{
					#line 70 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
					int _g = 0;
					#line 70 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
					global::Array<object> _g1 = this.vvoices;
					#line 70 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
					while (( _g < _g1.length )){
						#line 70 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						global::nx3.PVoice vvoice = ((global::nx3.PVoice) (_g1[_g]) );
						#line 70 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						 ++ _g;
						#line 72 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						{
							#line 72 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
							int _g2 = 0;
							#line 72 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
							global::Array<object> _g3 = vvoice.getNotes();
							#line 72 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
							while (( _g2 < _g3.length )){
								#line 72 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
								global::nx3.PNote vnote = ((global::nx3.PNote) (_g3[_g2]) );
								#line 72 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
								 ++ _g2;
								#line 74 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
								global::haxe.lang.Null<int> npos = default(global::haxe.lang.Null<int>);
								#line 74 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
								{
									#line 74 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
									global::IMap<object, int> this1 = vvoice.getNotePositions();
									#line 74 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
									npos = this1.@get(vnote);
								}
								
								#line 75 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
								if ( ! (positionsMap.exists(npos.@value)) ) {
									#line 75 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
									positionsMap.@set(npos.@value, new global::Array<object>(new object[]{}));
								}
								
								#line 76 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
								((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (positionsMap.@get(npos.@value).@value) ))) ).push(vnote);
							}
							
						}
						
					}
					
				}
				
				#line 79 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				return positionsMap;
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				switch (hash){
					case 130086162:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						this.positionsMap = ((global::haxe.ds.IntMap<object>) (global::haxe.ds.IntMap<object>.__hx_cast<object>(((global::haxe.ds.IntMap) (@value) ))) );
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return @value;
					}
					
					
					case 1329310366:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						this.complexes = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return @value;
					}
					
					
					case 137270135:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						this.vvoices = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return @value;
					}
					
					
					case 1246880755:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						this.part = ((global::nx3.PPart) (@value) );
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				switch (hash){
					case 1700422983:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("calcPositionsMap") ), ((int) (1700422983) ))) );
					}
					
					
					case 130086162:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return this.positionsMap;
					}
					
					
					case 245970761:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("calcComplexes") ), ((int) (245970761) ))) );
					}
					
					
					case 1586644776:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getComplexes") ), ((int) (1586644776) ))) );
					}
					
					
					case 1329310366:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return this.complexes;
					}
					
					
					case 137270135:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return this.vvoices;
					}
					
					
					case 1246880755:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return this.part;
					}
					
					
					default:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				switch (hash){
					case 1700422983:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return this.calcPositionsMap();
					}
					
					
					case 245970761:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						this.calcComplexes(((global::haxe.ds.IntMap<object>) (global::haxe.ds.IntMap<object>.__hx_cast<object>(((global::haxe.ds.IntMap) (dynargs[0]) ))) ));
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						break;
					}
					
					
					case 1586644776:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return this.getComplexes();
					}
					
					
					default:
					{
						#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				return default(object);
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				baseArr.push("positionsMap");
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				baseArr.push("complexes");
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				baseArr.push("vvoices");
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				baseArr.push("part");
				#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
				{
					#line 13 "F:\\nx3\\src\\nx3\\PPartComplexesGenerator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/PPartComplexesGenerator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PPartComplexesGenerator extends haxe.lang.HxObject
{
	public    PPartComplexesGenerator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PPartComplexesGenerator(nx3.PPart part)
	{
		nx3.PPartComplexesGenerator.__hx_ctor_nx3_PPartComplexesGenerator(this, part);
	}
	
	
	public static   void __hx_ctor_nx3_PPartComplexesGenerator(nx3.PPartComplexesGenerator __temp_me105, nx3.PPart part)
	{
		__temp_me105.part = part;
		__temp_me105.vvoices = part.getVoices();
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PPartComplexesGenerator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PPartComplexesGenerator(((nx3.PPart) (arr.__get(0)) ));
	}
	
	
	public  nx3.PPart part;
	
	public  haxe.root.Array<nx3.PVoice> vvoices;
	
	public  haxe.root.Array<nx3.PComplex> complexes;
	
	public   haxe.root.Array<nx3.PComplex> getComplexes()
	{
		if (( this.complexes != null )) 
		{
			return this.complexes;
		}
		
		this.positionsMap = this.calcPositionsMap();
		this.calcComplexes(this.positionsMap);
		return this.complexes;
	}
	
	
	public   void calcComplexes(haxe.ds.IntMap<haxe.root.Array> positions)
	{
		this.complexes = new haxe.root.Array<nx3.PComplex>(new nx3.PComplex[]{});
		haxe.root.Array<java.lang.Object> poskeys = ((haxe.root.Array<java.lang.Object>) (((haxe.root.Array) (cx.MapTools.keysToArray(positions.keys())) )) );
		poskeys = ((haxe.root.Array<java.lang.Object>) (((haxe.root.Array) (cx.MapTools.sortarray(((haxe.root.Array<java.lang.Object>) (((haxe.root.Array) (poskeys) )) ))) )) );
		{
			int _g = 0;
			while (( _g < poskeys.length ))
			{
				int pos = ((int) (haxe.lang.Runtime.toInt(poskeys.__get(_g))) );
				 ++ _g;
				haxe.root.Array<nx3.PNote> vnotes = ((haxe.root.Array<nx3.PNote>) (((haxe.root.Array) (positions.get(((int) (pos) ))) )) );
				nx3.PComplex vcomplex = new nx3.PComplex(((nx3.PPart) (this.part) ), ((haxe.root.Array<nx3.PNote>) (vnotes) ), ((int) (pos) ));
				this.complexes.push(vcomplex);
			}
			
		}
		
	}
	
	
	public  haxe.ds.IntMap<haxe.root.Array> positionsMap;
	
	public   haxe.ds.IntMap<haxe.root.Array> calcPositionsMap()
	{
		haxe.ds.IntMap<haxe.root.Array> positionsMap = new haxe.ds.IntMap<haxe.root.Array>();
		{
			int _g = 0;
			haxe.root.Array<nx3.PVoice> _g1 = this.vvoices;
			while (( _g < _g1.length ))
			{
				nx3.PVoice vvoice = _g1.__get(_g);
				 ++ _g;
				{
					int _g2 = 0;
					haxe.root.Array<nx3.PNote> _g3 = vvoice.getNotes();
					while (( _g2 < _g3.length ))
					{
						nx3.PNote vnote = _g3.__get(_g2);
						 ++ _g2;
						java.lang.Object npos = null;
						{
							haxe.root.IMap<nx3.PNote, java.lang.Object> this1 = vvoice.getNotePositions();
							npos = this1.get(vnote);
						}
						
						if ( ! (positionsMap.exists(((int) (haxe.lang.Runtime.toInt(npos)) ))) ) 
						{
							positionsMap.set(((int) (haxe.lang.Runtime.toInt(npos)) ), ((haxe.root.Array) (new haxe.root.Array<nx3.PNote>(new nx3.PNote[]{})) ));
						}
						
						((haxe.root.Array<nx3.PNote>) (((haxe.root.Array) (positionsMap.get(((int) (haxe.lang.Runtime.toInt(npos)) ))) )) ).push(vnote);
					}
					
				}
				
			}
			
		}
		
		return positionsMap;
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef559 = true;
			switch (field.hashCode())
			{
				case 34951314:
				{
					if (field.equals("positionsMap")) 
					{
						__temp_executeDef559 = false;
						this.positionsMap = ((haxe.ds.IntMap<haxe.root.Array>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3433459:
				{
					if (field.equals("part")) 
					{
						__temp_executeDef559 = false;
						this.part = ((nx3.PPart) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -1402927778:
				{
					if (field.equals("complexes")) 
					{
						__temp_executeDef559 = false;
						this.complexes = ((haxe.root.Array<nx3.PComplex>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 835229047:
				{
					if (field.equals("vvoices")) 
					{
						__temp_executeDef559 = false;
						this.vvoices = ((haxe.root.Array<nx3.PVoice>) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef559) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef560 = true;
			switch (field.hashCode())
			{
				case -643217529:
				{
					if (field.equals("calcPositionsMap")) 
					{
						__temp_executeDef560 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("calcPositionsMap"))) );
					}
					
					break;
				}
				
				
				case 3433459:
				{
					if (field.equals("part")) 
					{
						__temp_executeDef560 = false;
						return this.part;
					}
					
					break;
				}
				
				
				case 34951314:
				{
					if (field.equals("positionsMap")) 
					{
						__temp_executeDef560 = false;
						return this.positionsMap;
					}
					
					break;
				}
				
				
				case 835229047:
				{
					if (field.equals("vvoices")) 
					{
						__temp_executeDef560 = false;
						return this.vvoices;
					}
					
					break;
				}
				
				
				case -1817150839:
				{
					if (field.equals("calcComplexes")) 
					{
						__temp_executeDef560 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("calcComplexes"))) );
					}
					
					break;
				}
				
				
				case -1402927778:
				{
					if (field.equals("complexes")) 
					{
						__temp_executeDef560 = false;
						return this.complexes;
					}
					
					break;
				}
				
				
				case -1412532568:
				{
					if (field.equals("getComplexes")) 
					{
						__temp_executeDef560 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getComplexes"))) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef560) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef561 = true;
			switch (field.hashCode())
			{
				case -643217529:
				{
					if (field.equals("calcPositionsMap")) 
					{
						__temp_executeDef561 = false;
						return this.calcPositionsMap();
					}
					
					break;
				}
				
				
				case -1412532568:
				{
					if (field.equals("getComplexes")) 
					{
						__temp_executeDef561 = false;
						return this.getComplexes();
					}
					
					break;
				}
				
				
				case -1817150839:
				{
					if (field.equals("calcComplexes")) 
					{
						__temp_executeDef561 = false;
						this.calcComplexes(((haxe.ds.IntMap<haxe.root.Array>) (dynargs.__get(0)) ));
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef561) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			
		}
		
		return null;
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("positionsMap");
		baseArr.push("complexes");
		baseArr.push("vvoices");
		baseArr.push("part");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


